server/src/main/java/org/cloudfoundry/identity/uaa/provider/saml/Saml2BearerGrantAuthenticationConverter.java

```diff
@@ -184,36 +186,26 @@ static Converter<OpenSaml4AuthenticationProvider.AssertionToken, AbstractAuthent
 
     @Override
     public Authentication convert(HttpServletRequest request) throws AuthenticationException {
-        RelyingPartyRegistration relyingPartyRegistration = relyingPartyRegistrationResolver.resolve(request, null);
 
         String serializedAssertion = request.getParameter("assertion");
-        byte[] decodedAssertion = Saml2Utils.samlDecode(serializedAssertion);
+        byte[] decodedAssertion = Saml2Utils.samlBearerDecode(serializedAssertion);
         String assertionXml = new String(decodedAssertion, StandardCharsets.UTF_8);
 
         Assertion assertion = parseAssertion(assertionXml);
+        RelyingPartyRegistration relyingPartyRegistration = relyingPartyRegistrationResolver.resolve(request, getIssuer(assertion));
+        IdentityProvider<SamlIdentityProviderDefinition> idp = retrieveSamlIdpSamlIdentityProvider(relyingPartyRegistration.getRegistrationId());
```

Comment:
this change here simply ensures we have the IDP (and active) before we validate the signature.